### ISSUE TITLE
NEW 24 Hour Trade Order

### ISSUE THREAD

**Issue by eitanebench** [jmfernandes/robin_stocks#409]
Hi,
Has anyone figured out how to place an hour good in the new 24 hour trading period?

**Comment by henryzhangpku** [comment]
Yes this feature has been included in my PR https://github.com/jmfernandes/robin_stocks/issues/403 and has already been merged . You should be able to access 24hour market trading with this new version right away.

**Comment by henryzhangpku** [comment]
See my 24 hours market test example here https://github.com/jmfernandes/robin_stocks/issues/401#issuecomment-1609013310

**Comment by eitanebench** [comment]
order() got an unexpected keyword argument 'market_hours
still getting the above error, even though I see robin_stocks has it in the order function

**Comment by henryzhangpku** [comment]
I am in Pacific Time. Hope this example helps. 
--
    #which hours ? 
    extendHour = False
    market_hours='regular_hours'

    if dt.time(6, 30, 0) > dt.datetime.now().time() or dt.datetime.now().time() > dt.time(13, 0, 0):
        size = int(size)
        extendHour = True
        market_hours= 'all_day_hours'
    if order_amount > 0:

        r.order_buy_fractional_by_price(cbid, amountInDollars=round(
            USD * order_amount * close_ratio, 1), account_number=account_number, timeInForce="gfd", extendedHours=extendHour, jsonify=True, market_hours=market_hours)

**Comment by eitanebench** [comment]
Thanks.
Just realized its only on my chromebook using jupyter notebook but in google colab it works fine.

**Comment by joeld1** [comment]
Any updates on this issue? I upgraded to 3.0.6 using poetry but wasn't able to place an order_sell_limit order with `market_hours = 'all_day_hours'` passed in.

**Comment by eitanebench** [comment]
I haven't been able to solve this issue on my Chromebook so I switched to a windows 11 PC and it works fine with the same code.
Seems to be an issue within Linux in the Chromebook.

**Comment by joeld1** [comment]
Huh, I was using a mac for this

**Comment by joeld1** [comment]
Huh, `market_hours` isn't available for `order_buy_limit` ? Is this how it's supposed to be? I thought we could perform limit orders on the 24hr market?

**Comment by doctorcolossus** [comment]
@joeld1, good point. You might want to open a new issue about that or a pull request to fix it. The fix would be easy.

`order_buy_limit()` does indeed seem to be missing a `market_hours` argument, but if you look at it's body, all it does is just wrap a call to `order()`:

```python
def order_buy_limit(symbol,
                    quantity,
                    limitPrice,
                    account_number=None,
                    timeInForce='gtc',
                    extendedHours=False,
                    jsonify=True):

    return order(symbol,
                 quantity,
                 "buy",
                 limitPrice,
                 None,
                 account_number,
                 timeInForce,
                 extendedHours,
                 jsonify)
```

`order()` does provide that argument:

```python
def order(symbol,
          quantity,
          side,
          limitPrice=None,
          stopPrice=None,
          account_number=None,
          timeInForce='gtc',
          extendedHours=False,
          jsonify=True,
          market_hours='regular_hours'):  # ⬅️
```

So you'd just need to update the wrapper functions to also accept that argument and pass it to `order()`.

**Comment by doctorcolossus** [comment]
24-hour orders have been working fine for me via `order()` for quite some time.

This issue can be closed, can't it?
